base/src/main/java/io/spine/util/Text.java

```diff
@@ -0,0 +1,206 @@
+     *
+     * @throws IllegalArgumentException
+     *          if any of the lines contains the {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public Text(Iterable<String> lines) {
+        checkNotNull(lines);
+        this.lines = ImmutableList.copyOf(lines);
+    }
+
+    /**
+     * Creates a new instance splitting the given text into lines.
+     */
+    public Text(String text) {
+        this(split(text));
+    }
+
+    /**
+     * Creates a new instance with the given lines.
+     *
+     * <p>All the given lines must not contain a {@linkplain System#lineSeparator()
+     * line separator}.
+     *
+     * @throws IllegalArgumentException
+     *          if any of the lines contains the {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public Text(String[] lines) {
+        this(ImmutableList.copyOf(lines));
+    }
+
+    /**
+     * Creates a new list with the given lines.
+     *
+     * <p>All the given lines must not contain a {@linkplain System#lineSeparator()
+     * line separator}.
+     *
+     * @throws IllegalArgumentException
+     *          if any of the lines contains the {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public static Text of(String... lines) {
+        checkNotNull(lines);
+        return new Text(lines);
+    }
+
+    /**
+     * Obtains a read-only view of the text lines.
+     */
+    public List<String> lines() {
+        return lines;
+    }
+
+    /**
+     * Tells if the given string is contained by any of the text lines.
+     *
+     * @param s
+     *        the string to find. Must not contain a {@linkplain System#lineSeparator()
+     *        line separator}.
+     * @return {@code true} if at least one line contains the given string, {@code false} otherwise
+     * @throws IllegalArgumentException
+     *          if the given string contains the {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public boolean contains(String s) {
+        checkArgument(!s.contains(lineSeparator()));
+        var result = lines.stream().anyMatch(line -> line.contains(s));
+        return result;
+    }
+
+    /**
+     * Obtains the text with joined lines separated by {@linkplain System#lineSeparator()
+     * line separator}.
+     */
+    @Override
+    public String toString() {
+        return join(lines);
+    }
+
+    @Override
+    public Iterator<String> iterator() {
+        return lines.stream().iterator();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Text)) {
+            return false;
+        }
+        var strings = (Text) o;
+        return Objects.equals(lines, strings.lines);
+    }
+
+    @Override
+    public int hashCode() {
+        return lines.hashCode();
+    }
+
+    /*
+     * Static utilities. Please prefer instance methods of {@code Text}.
```

Comment:
Please format the section according to Spine guidelines. 

Or move this recommendation to some other place, which will be reflected in the rendered Javadocs.